expression/aggregation/base_func.go

```diff
@@ -155,6 +160,7 @@ func (a *baseFuncDesc) typeInfer4Avg(ctx sessionctx.Context) {
 			a.RetTp.Decimal = mathutil.Min(a.Args[0].GetType().Decimal+types.DivFracIncr, mysql.MaxDecimalScale)
 		}
 		a.RetTp.Flen = mysql.MaxDecimalWidth
+		logutil.Logger(context.Background()).Warn("set field type Flen to MaxDecimalWidth", zap.String("sql", ctx.GetSessionVars().StmtCtx.OriginalSQL), zap.Error(expression.ErrUnknown.GenWithStackByArgs()))
```

Comment:
> zap.Error(expression.ErrUnknown.GenWithStackByArgs())

The stack cannot be printed in this way, you can check the log here by a simple statement:
```
select sum(a) from t;
```
whereas `a` is of type `DECIMAL`

I found that using `zap.Stack("stack")` just works.